File: frontend/types/job.ts
export type JobStatus = "draft" | "open" | "closed" | "archived";

export interface Job {
  id: string;
  title: string;
  department: string | null;
  description: string;
  requirements: string | null;
  skills: string[];
  experience_level: string | null;
  salary_min: number | null;
  salary_max: number | null;
  location: string | null;
  employment_type: string | null;
  status: JobStatus;
  created_by_id: string;
  created_at: string;
  updated_at: string;
}

export interface JobCreatePayload {
  title: string;
  department?: string;
  description: string;
  requirements?: string;
  skills?: string[];
  experience_level?: string;
  salary_min?: number;
  salary_max?: number;
  location?: string;
  employment_type?: string;
}

export type JobUpdatePayload = Partial<JobCreatePayload> & { status?: JobStatus };

File: backend/app/schemas/job.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.db.postgres.models import JobStatus


class JobBase(BaseModel):
    title: str = Field(..., min_length=2, max_length=255)
    department: str | None = None
    description: str = Field(..., min_length=10)
    requirements: str | None = None
    skills: list[str] = Field(default_factory=list)
    experience_level: str | None = None
    salary_min: int | None = Field(default=None, ge=0)
    salary_max: int | None = Field(default=None, ge=0)
    location: str | None = None
    employment_type: str | None = None


class JobCreate(JobBase):
    pass


class JobUpdate(BaseModel):
    title: str | None = None
    department: str | None = None
    description: str | None = None
    requirements: str | None = None
    skills: list[str] | None = None
    experience_level: str | None = None
    salary_min: int | None = None
    salary_max: int | None = None
    location: str | None = None
    employment_type: str | None = None
    status: JobStatus | None = None


class JobOut(JobBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    status: JobStatus
    created_by_id: str
    created_at: datetime
    updated_at: datetime
